cli/src/main/java/org/openjdk/skara/cli/GitSync.java

```diff
@@ -0,0 +1,164 @@
+                  .optional(),
+            Switch.shortcut("m")
+                  .fullname("mercurial")
+                  .helptext("Force use of mercurial")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("verbose")
+                  .helptext("Turn on verbose output")
+                  .optional(),
+            Switch.shortcut("")
+                  .fullname("debug")
+                  .helptext("Turn on debugging output")
+                  .optional(),
+            Switch.shortcut("v")
+                  .fullname("version")
+                  .helptext("Print the version of this tool")
+                  .optional()
+        );
+
+        var inputs = List.of(
+            Input.position(0)
+                 .describe("REMOTE")
+                 .singular()
+                 .optional()
+        );
+
+        var parser = new ArgumentParser("git sync", flags, inputs);
+        var arguments = parser.parse(args);
+
+        if (arguments.contains("version")) {
+            System.out.println("git-sync version: " + Version.fromManifest().orElse("unknown"));
+            System.exit(0);
+        }
+
+        if (arguments.contains("verbose") || arguments.contains("debug")) {
+            var level = arguments.contains("debug") ? Level.FINER : Level.FINE;
+            Logging.setup(level);
+        }
+
+        var cwd = Paths.get("").toAbsolutePath();
+        var repo = Repository.get(cwd).orElseThrow(() ->
+                die("error: no repository found at " + cwd.toString())
+        );
+
+        var remotes = repo.remotes();
+
+        String upstream = null;
+        if (arguments.at(0).isPresent()) {
+            upstream = arguments.at(0).asString();
+        } else {
+            var lines = repo.config("sync.remote");
+            if (lines.size() == 1 && remotes.contains(lines.get(0))) {
+                upstream = lines.get(0);
+            } else {
+                die("No remote provided to sync with");
+            }
+        }
+
+        var upstreamPullPath = remotes.contains(upstream) ?
+            Remote.toURI(repo.pullPath(upstream)) : URI.create(upstream);
+        var origin = "origin";
+        var originPushPath = Remote.toURI(repo.pushPath(origin));
+
+        var branches = new HashSet<String>();
+        if (arguments.contains("branches")) {
+            var requested = arguments.get("branches").asString().split(",");
+            for (var branch : requested) {
+                branches.add(branch.trim());
+            }
+        }
+
+        for (var branch : repo.remoteBranches(upstream)) {
+            var name = branch.name();
+            if (!branches.isEmpty() && !branches.contains(name)) {
+                System.out.println("Skipping branch " + name);
+                continue;
+            }
+            System.out.print("Syncing " + upstream + "/" + name + " to " + origin + "/" + name + "... ");
```

Comment:
Do you need to flush the PrintStream here?

Reply:
Probably, good idea, let me fix!